src/common/components/clusterConfiguration/utils.ts

```diff
@@ -66,15 +62,34 @@ export const getSubnetFromMachineNetworkCidr = (machineNetworkCidr?: string) =>
   return subnet?.address;
 };
 
+const serviceNetworksEqual = (array1: ServiceNetwork[], array2: ServiceNetwork[]) =>
+  array1.length === array2.length &&
+  array1.every((network) => array2.map((elem) => elem.cidr).includes(network.cidr));
+
+const clusterNetworksEqual = (array1: ClusterNetwork[], array2: ClusterNetwork[]) =>
+  array1.length === array2.length &&
+  array1.every((clusterNetwork) =>
+    array2.find(
+      (network) =>
+        network.cidr === clusterNetwork.cidr && network.hostPrefix === clusterNetwork.hostPrefix,
+    ),
+  );
+
 export const isAdvNetworkConf = (
   cluster: Cluster,
-  defaultNetworkSettings: ClusterDefaultConfig,
+  defaultNetworkValues: Partial<NetworkConfigurationValues>,
```

Comment:
I would prefer `Pick` instead of `Partial` as it gives better idea on what is actually expected